goth_tests/domain/ya-provider/test_provider_multi_activity.py

```diff
@@ -159,3 +163,106 @@ async def test_provider_single_simultaneous_activity(
 
         await requestor.terminate_agreement(agreement_id, None)
         await provider.wait_for_agreement_terminated()
+
+
+@pytest.mark.asyncio
+async def test_provider_recover_from_abandoned_task(
+    common_assets: Path,
+    config_overrides: List[Override],
+    log_dir: Path,
+):
+    """Tests providers' ability of terminating an abandoned task
+    and starting another one"""
+
+    nodes = [
+        {"name": "requestor-1", "type": "Requestor"},
+        {"name": "requestor-2", "type": "Requestor"},
+        {"name": "provider-1", "type": "Wasm-Provider", "use-proxy": True},
+    ]
+
+    config_overrides.append(("nodes", nodes))
+    runner, config = _create_runner(common_assets, config_overrides, log_dir)
+
+    async with runner(config.containers):
+        requestors = runner.get_probes(probe_type=RequestorProbe)
+        assert requestors
+        providers = runner.get_probes(probe_type=ProviderProbe)
+        assert providers
+
+        def build_demand(requestor, sleeper_task: bool = False):
+            task_package = (
+                wasi_sleeper_task_package if sleeper_task else wasi_task_package
+            )
+            return (
+                DemandBuilder(requestor)
+                .props_from_template(
+                    task_package.format(
+                        web_server_addr=runner.host_address,
+                        web_server_port=runner.web_server_port,
+                    )
+                )
+                .property("golem.srv.caps.multi-activity", True)
+                .property("golem.com.payment.debit-notes.accept-timeout?", 5)
+                .constraints(
+                    "(&(golem.com.pricing.model=linear)\
+                    (golem.srv.caps.multi-activity=true)\
+                    (golem.runtime.name=wasmtime))"
+                )
+                .build()
+            )
```

Comment:
What is the benefit of defining function inside other function over defining it on top level?

Reply:
It's next to other test-specific functions. Other tests wouldn't benefit from having it at top level